node/trieIterators/delegatedListProcessor.go

```diff
@@ -0,0 +1,212 @@
+	delegatorsList, err := dlp.getDelegatorsList(delegationSC)
+	if err != nil {
+		return err
+	}
+
+	var value *big.Int
+	for _, delegatorAddress := range delegatorsList {
+		value, err = dlp.getActiveFund(delegationSC, delegatorAddress)
+		if err != nil {
+			//delegatorAddress byte slice might not represent a real delegator address
+			continue
+		}
+
+		delegatorInfo, ok := delegatorsMap[string(delegatorAddress)]
+		if !ok {
+			delegatorInfo = &api.Delegator{
+				DelegatorAddress: delegatorAddress,
+				DelegatedTo:      make([]api.DelegatedValue, 0),
+				Total:            big.NewInt(0),
+			}
+
+			delegatorsMap[string(delegatorAddress)] = delegatorInfo
+		}
+
+		delegatorInfo.Total = big.NewInt(0).Add(delegatorInfo.Total, value)
+		delegatorInfo.DelegatedTo = append(delegatorInfo.DelegatedTo, api.DelegatedValue{
+			DelegationScAddress: delegationSC,
+			Value:               big.NewInt(0).Set(value),
+		})
+	}
+
+	return nil
+}
+
+func (dlp *delegatedListProc) getDelegatorsList(delegationSC []byte) ([][]byte, error) {
+	delegatorAccount, err := dlp.getAccount(delegationSC)
+	if err != nil {
+		return nil, fmt.Errorf("%w for delegationSC %s", err, hex.EncodeToString(delegationSC))
+	}
+
+	rootHash, err := delegatorAccount.DataTrie().RootHash()
+	if err != nil {
+		return nil, fmt.Errorf("%w for delegationSC %s", err, hex.EncodeToString(delegationSC))
+	}
+
+	ctx := context.Background()
+	chLeaves, err := delegatorAccount.DataTrie().GetAllLeavesOnChannel(rootHash, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	delegators := make([][]byte, 0)
+	for leaf := range chLeaves {
+		leafKey := leaf.Key()
+		if len(leafKey) != dlp.publicKeyConverter.Len() {
+			continue
+		}
+
+		delegators = append(delegators, leafKey)
+	}
+
+	return delegators, nil
+}
+
+func (dlp *delegatedListProc) getActiveFund(delegationSC []byte, delegator []byte) (*big.Int, error) {
+	scQuery := &process.SCQuery{
+		ScAddress:  delegationSC,
+		FuncName:   "getUserActiveStake",
+		CallerAddr: delegationSC,
+		CallValue:  big.NewInt(0),
+		Arguments:  [][]byte{delegator},
+	}
+
+	vmOutput, err := dlp.queryService.ExecuteQuery(scQuery)
+	if err != nil {
+		return nil, err
+	}
+	if vmOutput.ReturnCode != vmcommon.Ok {
+		return nil, fmt.Errorf("%w, error: %v message: %s", epochStart.ErrExecutingSystemScCode, vmOutput.ReturnCode, vmOutput.ReturnMessage)
+	}
+
+	if len(vmOutput.ReturnData) != 1 {
+		return nil, fmt.Errorf("%w, getActiveFund function should have at least one value", epochStart.ErrExecutingSystemScCode)
+	}
+
+	value := big.NewInt(0).SetBytes(vmOutput.ReturnData[0])
+
+	return value, nil
+}
+
+func (dlp *delegatedListProc) mapToSlice(mapDelegators map[string]*api.Delegator) []*api.Delegator {
+	keys := make([]string, 0, len(mapDelegators))
+	for key := range mapDelegators {
+		keys = append(keys, key)
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
```

Comment:
why do you need to sort the elements in the slice? you use them as keys in a map.
Am I missing something?

Reply:
Just wanted to be consistent between calls (same call will produce exactly the same output for the same data). Might be dropped.